Mathlib/Condensed/Discrete/Colimit.lean

```diff
@@ -0,0 +1,373 @@
+  val := lanPresheaf (locallyConstantPresheaf X)
+  cond := by
+    rw [Presheaf.isSheaf_of_iso_iff (lanPresheafNatIso fun _ ↦ isColimit _ _)]
+    exact ((functor.{u, u+1} (hs := fun _ _ _ ↦ ((effectiveEpi_tfae _).out 0 2).mp)).obj X).cond
+
+/-- `lanPresheaf (locallyConstantPresheaf X)` as a condensed set. -/
+def lanCondensedSet (X : Type (u+1)) : CondensedSet.{u} :=
+  (ProfiniteCompHaus.equivalence _).functor.obj (lanSheafProfinite X)
+
+variable (F : Profinite.{u}ᵒᵖ ⥤ Type (u+1))
+
+/--
+The functor which takes a finite set to the set of maps into `F(*)` for a presheaf `F` on
+`Profinite`.
+-/
+@[simps]
+def finYoneda : FintypeCat.{u}ᵒᵖ ⥤ Type (u+1) where
+  obj X := X.unop → F.obj (toProfinite.op.obj ⟨of PUnit.{u+1}⟩)
+  map f g := g ∘ f.unop
+
+/-- `locallyConstantPresheaf` restricted to finite sets is isomorphic to `finYoneda F`. -/
+def locallyConstantIsoFinYoneda :
+    toProfinite.op ⋙ (locallyConstantPresheaf (F.obj (toProfinite.op.obj ⟨of PUnit.{u+1}⟩))) ≅
+    finYoneda F :=
+  NatIso.ofComponents fun Y ↦ {
+    hom := fun f ↦ f.1
+    inv := fun f ↦ ⟨f, @IsLocallyConstant.of_discrete _ _ _ ⟨rfl⟩ _⟩ }
+
+/-- A finite set as a coproduct cocone in `Profinite` over itself. -/
+def fintypeCatAsCofan (X : FintypeCat) : Cofan (fun (_ : X) ↦ toProfinite.obj (of (PUnit.{u+1}))) :=
+  Cofan.mk (toProfinite.obj X) (fun x ↦ toProfinite.map (fun _ ↦ x))
+
+/-- A finite set is the coproduct of its points in `Profinite`. -/
+def fintypeCatAsCofanIsColimit (X : FintypeCat.{u}) : IsColimit (fintypeCatAsCofan X) :=
+  mkCofanColimit _ (fun t ↦ ⟨fun x ↦ t.inj x PUnit.unit, continuous_bot⟩) (by aesop)
+    (fun _ _ h ↦ by ext x; exact ContinuousMap.congr_fun (h x) _)
+
+variable [PreservesFiniteProducts F]
+
+/-- Auxiliary definition for `isoFinYoneda`. -/
+@[simps!]
+def isoFinYonedaComponents (X : FintypeCat.{u}) :
+    F.obj ((toProfinite.{u}.op.obj ⟨X⟩)) ≅ (X → F.obj (toProfinite.op.obj ⟨of PUnit.{u+1}⟩)) :=
+  (isLimitFanMkObjOfIsLimit F _ _
+    (Cofan.IsColimit.op (fintypeCatAsCofanIsColimit X))).conePointUniqueUpToIso
+      (Types.productLimitCone.{u, u+1} fun _ ↦ F.obj (toProfinite.op.obj ⟨of PUnit.{u+1}⟩)).2
+
+/--
+The restriction of a finite product preserving presheaf `F` on `Profinite` to the category of
+finite sets is isomorphic to `finYoneda F`.
+-/
+def isoFinYoneda : toProfinite.op ⋙ F ≅ finYoneda F :=
+  NatIso.ofComponents (fun X ↦ isoFinYonedaComponents F X.unop) fun _ ↦ by
+    simp only [comp_obj, op_obj, finYoneda_obj, Functor.comp_map, op_map]
+    ext
+    simp only [types_comp_apply, isoFinYonedaComponents_hom, finYoneda_map, op_obj,
+      Function.comp_apply, Types.productLimitCone, const_obj_obj, fintypeCatAsCofan, Cofan.mk_pt,
+      cofan_mk_inj, Fan.mk_pt, Fan.mk_π_app, ← FunctorToTypes.map_comp_apply]
+    rfl
+
+/--
+A presheaf `F`, which takes a profinite set written as a cofiltered limit to the corresponding
+colimit, is isomorphic to the presheaf `LocallyConstant - F(*)`.
```

Comment:
```suggestion
colimit, is isomorphic to the presheaf `LocallyConstant _ F(*)`.
```
Or did you mean something else?

Reply:
The `-` was intended as a "blank". So changed to `LocallyConstant · F(*)`.